cartography/intel/crxcavator.py

```diff
@@ -0,0 +1,210 @@
+            'size': data['webstore'].get('size'),
+            'support_site': data['webstore'].get('support_site'),
+            'users': data['webstore'].get('users'),
+            'website': data['webstore'].get('website'),
+            'type': data['webstore'].get('type'),
+            'price': data['webstore'].get('price')
+        })
+    logger.info('Ingesting {} extensions'.format(len(extensions)))
+    session.run(ingestion_cypher, ExtensionsData=extensions, UpdateTag=update_tag)
+
+
+def ingest_users_extensions(session, update_tag):
+    """
+    Ingests the extension to user mapping details into Neo4J
+    :param session: Neo4J session object for server communication
+    :param update_tag: Timestamp used to determine data freshness
+    :return: None
+    """
+    user_extensions_json = get_users_extensions()
+    if not user_extensions_json:
+        logger.error('Failed to get user data from CRXcavator')
+        return
+    # todo: once G Suite ingestion is created, move first MERGE to that process
+    user_ingestion_cypher = """
+    UNWIND {Users} as user_email
+    MERGE (user:GSuiteUser{id: user_email, email: user_email})
+    ON CREATE SET
+    user.firstseen = timestamp()
+    SET user.lastupdated = {UpdateTag}
+    """
+    # todo: add relationship to computer node when JAMF data is moved to Cartography
+    extension_ingestion_cypher = """
+    UNWIND {ExtensionsUsers} as extension_user
+    MATCH (user:GSuiteUser{email: extension_user.user}),(ext:ChromeExtension{id:extension_user.id})
+    MERGE (user)-[r:INSTALLS]->(ext)
+    ON CREATE SET
+    r.firstseen = timestamp()
+    SET r.lastupdated = {UpdateTag}
+    """
+
+    user_extensions = user_extensions_json.items()
+    users_set = set()
+    extensions_by_user = []
+    for extension in user_extensions:
+        for details in extension[1].items():
+            for user in details[1]['users']:
+                users_set.add(user)
+                extensions_by_user.append({
+                    'id': "{0}|{1}".format(extension[0], details[0]),
+                    'user': user})
+    users = list(users_set)
+    logger.info('Ingesting {} users'.format(len(users)))
+    session.run(user_ingestion_cypher, Users=users, UpdateTag=update_tag)
+    logger.info('Ingesting {} user->extension relationships'.format(len(extensions_by_user)))
+    session.run(extension_ingestion_cypher, ExtensionsUsers=extensions_by_user, UpdateTag=update_tag)
+
+
+def start_extension_ingestion(session, config):
+    """
+    If this module is configured, perform ingestion of CRXcavator data. Otherwise warn and exit
+    :param session: Neo4J session for database interface
+    :param config: Neo4J server URI and Update tag for data freshness
+    :return: None
+    """
+    if not CRXcavator_API_BASE_URL or not CRXcavator_API_KEY:
+        logger.warning('CRXcavator import is not configured - skipping this module')
+        return None
```

Comment:
No need to `return None`.  Just `return` to break out.

Reply:
done